src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -662,6 +668,20 @@ protected void maybeCallbackAfterSaveWithQueries(List<?> queries, IndexCoordinat
 
 	// endregion
 
+	protected void updateIndexedObjectsWithQueries(List<?> queries,
+			List<IndexedObjectInformation> indexedObjectInformations) {
+		for (int i = 0; i < queries.size(); i++) {
+			Object query = queries.get(0);
```

Comment:
if you use `i` instead of `0` it works and the tests will be green. 😉

Reply:
Oops, that's embarassing :) Night coding...

I've fixed this and also updated my tests to catch it as well.